### DISCUSSION TITLE
RP2040(Raspberry Pie Pico) will support?

### DISCUSSION THREAD

**Discussion by Shootingmaker** [FortySevenEffects/arduino_midi_library#227] [General]
I'm working on running USB-MDI on the RP2040.
Currently, I get the correct error as below. "Unsupported architecture" 

Will we add RP2040 to support in the future?

`WARNING: library MIDIUSB claims to run on avr, sam, samd architecture(s) and may be incompatible with your current board which runs on mbed_rp2040 architecture(s).
In file included from C:\Users\Shoo\Documents\Arduino\libraries\USB-MIDI\src/USB-MIDI.h:26:0,
                 from c:\Users\Shoo\AppData\Local\Temp\.arduinoIDE-unsaved202165-300-1nqqzut.6ngw\midi_test\midi_test.ino:20:
C:\Users\Shoo\Documents\Arduino\libraries\MIDIUSB\src/MIDIUSB.h:18:2: error: #error MIDIUSB can only be used with an USB MCU.
 #error MIDIUSB can only be used with an USB MCU.
  ^~~~~
In file included from C:\Users\Shoo\Documents\Arduino\libraries\USB-MIDI\src/USB-MIDI.h:26:0,
                 from c:\Users\Shoo\AppData\Local\Temp\.arduinoIDE-unsaved202165-300-1nqqzut.6ngw\midi_test\midi_test.ino:20:
C:\Users\Shoo\Documents\Arduino\libraries\MIDIUSB\src/MIDIUSB.h:78:2: error: #error "Unsupported architecture"
 #error "Unsupported architecture"
  ^~~~~
Compilation error: Error: 2 UNKNOWN: exit status 1`

**Comment by lathoub** [comment]
Assuming you are using https://github.com/lathoub/Arduino-USBMIDI, in its turn depending on https://github.com/arduino-libraries/MIDIUSB and then on PluggableUSB. PluggableUSB only targets boards with native USB capabilities, and only supports these boards (eg. Leonardo, Micro, Due, Zero and so on)

No clue how RP2040 exposes USB capabilities, I will look it
